### ISSUE TITLE
feature creep

### ISSUE THREAD

**Issue by kbrock** [nithinbekal/actioncable-chat-example#12]
I plan on doing the following in my local prototype.
Which of these do you think will add to this example and not just make the code more complicated?

UNKNOWN:
- [ ] add a user object
- [ ] add to user presence
- [ ] add to user metadata like an icon
- [ ] add authentication to user object
  - [ ] simple password
  - [ ] devise
  - [ ] github

YES:
- [ ] add to message a target user for 1-1 chatting. [make this like a channel]
- [ ] add to message a target channel [for multiple rooms]
- [ ] add to ui a second pane with list of users.

NO:
- [ ] add to message a message type. (e.g. tweet, image)
- [ ] add to message meta data. (use postgres hext?)

**Comment by nithinbekal** [comment]
List of users definitely adds a lot of value, and also multiple channels. :+1:

By "adding message to target user", do you mean 1-1 chatting? That makes sense to have if we're adding multiple channels. (Maybe even figure out a way to implicitly make each 1-1 conversation a channel, perhaps?)

The others seem to be features that don't specifically add to the understanding of ActionCable (correct me if I've misunderstood there), so they are probably adding a bit too much complexity? I'm not really sure about this, though.

**Comment by kbrock** [comment]
Thanks @nithinbekal - ok, pass # 2

I don't like most of the examples out there because they are too simple.
But once an example gets sufficiently complex, it becomes worthless. 

So I appreciate another person's perspective. Hopefully with 2 brains, this can not add bloat.

While I do want to use this example to explore metadata in a chat and displaying the results in multiple panes, that would detract from the basics of active_channel.

**Comment by nithinbekal** [comment]
@kbrock Could you elaborate on what you meant by adding metadata to messages? I'm not fully clear about what it would look like.

**Comment by kbrock** [comment]
@nithinbekal Was thinking hash type field, probably a [hstore](http://www.postgresql.org/docs/9.0/static/hstore.html) column where you could store various pieces of information:
- `type` field like `"tweet"`, `"image"`, `"survey"`, `"build_status"` that will change the way a message is displayed.
- `tags` containing `#list #of #hashtags`
- any other field that works for the current implementer

**Comment by nithinbekal** [comment]
Ah, now I understand. That seems tempting to include and I'm ok with including those and making this example a bit more interesting. :) Maybe we should explore how we can isolate the non-cable code from the basic example.

If we're going ahead with more complex code, I will try to write a code walkthrough in the README so the people looking at this will have a high level overview of the code while reading it.

**Comment by huzefahuzu** [comment]
@nithinbekal  can you please elaborate on how i can i use our code to display the chats in my application without refreshing the page ? . the link to the repo is here : https://github.com/huzefahuzu/ChatApp

**Comment by nithinbekal** [comment]
@huzefahuzu This repo contains the code for this chat tutorial that I wrote: http://nithinbekal.com/posts/rails-action-cable/

You can refer to it to walk through the code.

**Comment by huzefahuzu** [comment]
@nithinbekal Yes i know that , but when i run your application on my machine , the browser just keep on loading and the application never starts . Also , the logs also doesn't get raised for any redirecting or rendering in the terminal . So , please let me know how to recover from this.

**Comment by nithinbekal** [comment]
@huzefahuzu It's very difficult to figure out what's wrong with such limited information. Can you explain what steps you have tried so far?

**Comment by huzefahuzu** [comment]
@nithinbekal  Thnx but i have sorted that out and it's running now . The steps you have mentioned in your tutorial is for 2 users . What if i have a list of several users and by clicking on any of i can start chatting ? That's similar to facebook chat . Also this thing i have tried to implement in my chat application [here](https://github.com/huzefahuzu/ChatApp) . But in my app , the messages doesn't get updated if a new message arrives , it does after the page refresh . That's the reason i am interested in your approach so that we can extend the functionality in my chat applicaton . So please do checkout my repo and let me know the suggestions. Thank you :)
